### PR TITLE
[-] Technical - Improve errors handling

### PR DESCRIPTION
**Pull request by root-io**

### CODE REVIEW

File: src/services/authenticator.js [hunk 1]
```diff
@@ -9,13 +9,13 @@ function Authenticator() {
...
       return fs.readFileSync(`${os.homedir()}/.forestrc`);
-    } catch (e) {
+    } catch (error) {
       return null;
     }
   };
 
   this.login = (config) => {
-    agent
+    return agent
```
Comment (root-io) [inline src/services/authenticator.js]: It was not returned properly

File: src/commands/environments/create.js [hunk 2]
```diff
@@ -28,7 +29,8 @@ class CreateCommand extends Command {
           return;
         }
       }
-      this.error(error, { exit: 1 });
+      catchUnexpectedError(error);
```
Comment (VincentMolinie) [inline src/commands/environments/create.js]: `this.error()` is differrent than `console.error()` + `this.exit()` ! Please use `this.error()`.
Furthermore, catchUnexpectedError is qlways followed by exit(1). I think you should include the exit inside the catchUnexpectedError

File: src/utils/index.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const bugsnag = require('@bugsnag/js');
+const { BUGSNAG_API_KEY } = require('../../config/config');
+
+const bugsnagClient = bugsnag(BUGSNAG_API_KEY);
+
+function catchUnexpectedError(error) {
```
Comment (ArnaudValensi) [inline src/utils/index.js]: You can call it `logError(error, options)`. 
To do the same thing than `this.error(...)`, you can include `const Errors = require("@oclif/errors");` and call `Errors.error(input, options);`.

So then you can decide to exit or not by doing:
```
logError('Error message', { exit: 1 });
```

File: src/commands/environments/create.js (hunk 2 above)
Comment (ArnaudValensi) [reply src/commands/environments/create.js]: I left a comment, see below.

**Comment by root-io** [comment]
[Toolbelt - Add bugsnag](https://trello.com/c/Hr6Rpo09/3218-toolbelt-add-bugsnag)

**Comment by rap2hpoutre** [comment]
I guess this PR would be considered too broad with our new quality requirements. It includes at least 4 topics (unless I mistake):
 - adding bugsnag
 - changing deploy process
 - adding eslint rules
 - refactoring/changing log process

I think we wont loose anything that would be impossible to reimplement by closing this PR. WDYT @VincentMolinie? (since you review this PR)

**Comment by arnaudbesnier** [comment]
Old contribution, the codebase changed a lot, let's close it.

**Pull request closed without merging**